### PR TITLE
[FEAT] Segment Integration

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_

_1. What is the reason for the change?_
Integrate a new Analytics tool to improve data management

_2. What is the improvement/solution?_
Integrate Segment into the Mobile app

For more details refer to https://github.com/MetaMask/mobile-planning/issues/372

**Screenshots/Recordings**

Not Applicable

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/372

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/core/Analytics/MetaMetrics.constants.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const METAMETRICS_ANONYMOUS_ID = '0x0000000000000000';
+
+export const SEGMENT_REGULATIONS_ENDPOINT =
+  'https://platform.segmentapis.com/v1beta/workspaces/myworkspace/regulations';
```
Comment (gantunesr) [inline app/core/Analytics/MetaMetrics.constants.ts:4]: Update to delete endpoint

File: patches/@segment+analytics-react-native+2.7.0.patch [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+     this.appState = nextAppState;
+   }
+ 
+-  reset(resetAnonymousId: boolean = true) {
+-    const anonymousId =
+-      resetAnonymousId === true
+-        ? getUUID()
+-        : this.store.userInfo.get().anonymousId;
++  reset() {
++    const anonymousId = '0x0000000000000000';
```
Comment (gantunesr) [inline patches/@segment+analytics-react-native+2.7.0.patch:29]: Pass anon ID as argument

File: patches/@segment+analytics-react-native+2.7.0.patch [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+diff --git a/node_modules/@segment/analytics-react-native/src/client.tsx b/node_modules/@segment/analytics-react-native/src/client.tsx
+index 3309986..da1109e 100644
+--- a/node_modules/@segment/analytics-react-native/src/client.tsx
++++ b/node_modules/@segment/analytics-react-native/src/client.tsx
+@@ -67,6 +67,6 @@ export const useAnalytics = (): ClientMethods => {
+     flush: async () => client.flush(),
+     group: async (...args) => client.group(...args),
+     alias: async (...args) => client.alias(...args),
+-    reset: async (...args) => client.reset(...args),
++    reset: async () => client.reset(),
```
Comment (gantunesr) [inline patches/@segment+analytics-react-native+2.7.0.patch:42]: Pass anon ID as argument

File: app/core/Analytics/MetaMetrics.events.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  name,
+  anonymous: anonymous || false,
+});
+
+const MetaMetricsEvents = {
+  WALLET: {
+    OPEN: generateOpt('Wallet Opened'),
+    TOKEN_ADDED: generateOpt('Token Added'),
+    COLLECTIBLE_ADDED: generateOpt('Collectible Added'),
+  },
```
Comment (gantunesr) [inline app/core/Analytics/MetaMetrics.events.ts:18]: Delete test events

File: app/components/hooks/MetaMetrics/useMetaMetrics.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+const useMetaMetrics = () => {
+  const trackEventHook = useCallback(
+    (event: IMetaMetricsEvent, params: JsonMap = {}) => {
+      const { name, anonymous } = event;
+      if (anonymous) {
+        MetaMetrics.trackEvent(name, true, params);
+      } else {
+        MetaMetrics.trackEvent(name, false, params);
+      }
```
Comment (gantunesr) [inline app/components/hooks/MetaMetrics/useMetaMetrics.tsx:13]:
```suggestion
      MetaMetrics.trackEvent(name, anonymous, params);
```

**Review by Cal-L — APPROVED** [review]
Pair reviewed with some comments - LGTM once addressed!

**Pull request merged**